dot/rpc/modules/state.go

```diff
@@ -245,10 +245,29 @@ func (sm *StateModule) GetPairs(_ *http.Request, req *StatePairRequest, res *Sta
 	return nil
 }
 
-// Call isn't implemented properly yet.
-func (sm *StateModule) Call(_ *http.Request, _ *StateCallRequest, _ *StateCallResponse) error {
-	_ = sm.networkAPI
-	_ = sm.storageAPI
+// Call makes a call to the runtime.
+func (sm *StateModule) Call(_ *http.Request, req *StateCallRequest, res *StateCallResponse) error {
+	var blockHash = sm.blockAPI.BestBlockHash()
```

Comment:
Would it be more effficient to do this in the other order to avoid calling BestBlockHash unless we need to. I don't see a reason to make the call if it is provided

Reply:
It'd lead to a lot of pointer ops which I want to avoid. Do you know a better way to do this?

Reply:
maybe something like:
`var blockHash common.Hash{}
if req.Block != nil {
    blockHash = *req.Block
} else {
      //bs.getBlockHash
}`

Just an idea though, I imagine you can leave as is so feel free to resolve this issue at your own discretion